app/redux/cart.js

```diff
@@ -116,14 +116,29 @@ export const fetchCart = () => {
 export const newSessionCart = () => {
   return (dispatch, getState, { axios }) => {
     return axios.post('/api/cart/')
-      .then(response => {
-        return response.data
-      })
+      .then(response => response.data)
       .then(cart => dispatch(setCart(cart)))
       .catch(e => console.log(chalk.red(`Error in Redux thunk createSessionCart: ${e}`)))
   }
 };
 
+export const mergeCart = () => {
+  return (dispatch, getState, { axios }) => {
+    // grab id of user cart
+    return axios.put('/api/cart')
+      .then(userCart => {
+        // grab session cart
+        const { cart } = getState();
+        // loop through session cart's cart items and change cartId to user cart's id
+        for (let i = 0; i < cart.cartitems.length; i++) {
+          const cartItem = cart.cartitems[i]
+          return dispatch(updateCartItem(cartItem.id, { cartId: userCart.data.id }))
+        }
+      })
+      .catch(e => console.log('Error in Redux thunk mergeCart: ', e))
```

Comment:
on catch, can we have it return either the current cart or null or something instead of just erroring out? In case something goes wrong with the update or the PUT

Reply:
there is also a catch within the update/PUT thunk `updateCartItem`